### PR TITLE
Pb/jobs v2

### PR DESCRIPTION
**Pull request by pierrebeitz**
some small commits that happened while inspecting the code. feel free to cherry-pick as you like and close this PR.

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
File conflicts.

**Comment by pierrebeitz** [comment]
this is 

1. addressing some of @GeorgiSTodorov [findings](https://github.com/dcos/dcos-ui/pull/3599#issuecomment-466315215).

    > * CPU and GPU inputs glow red as soon as the form is opened
    > * CPU is red after entering invalid value and changing it to valid

    should be fixed. therefore added `novalidate` to the form to prevent browsers from additionally validating stuff.

    > * GPU is red when it is below 32 and cannot be changed below 32 using the arrows (is this intended?).

    good catch! i guess that's a copy-paste-error. can you second that @natmegs?

    *** > * When I am editing the form and I enter a new value for cpu, disk, mem and I click submit, the old values are visible for a split second. This might be misleading, because the users might see this and think that their new values weren't saved.

    we seem to reset the form's defaults values just a bit too early.

2. removing a lot of typescript errors. i had to refactor some old code that was not properly types. i also removed code that threw type errors but apparently was not used anymore (i checked by `grep`ing and by reading the commit messages - so i'm like 99% sure this code is not used anymore).

3. fix some types - e.g. `handleJSONErrorStateChange ` was claiming that is only ever needed to handle strings (but actually it also needs to handle nulls).

4. some minor refactorings to make things more concise.

File: plugins/jobs/src/js/types/JobRunStatusSummary.ts [hunk 1]
```diff
@@ -19,77 +19,14 @@ export function JobRunStatusSummaryTypeResolver(
   history: MetronomeJobHistorySummary
 ): JobRunStatusSummary {
   const summary = JobHistorySummaryTypeResolver(history);
-  let status = "N/A" as JobRunStatus;
```
[the remark this answers is not in the document]
Comment (pierrebeitz) [reply plugins/jobs/src/js/types/JobRunStatusSummary.ts]: ok, i removed that commit and will move it to a different PR.

**Comment by pierrebeitz** [comment]
> I'm ok with changing the react import but I wonder why I don't see any error/warning for this?

They are **currently** not visible when using webpack, as we introduced `thread-loader` in the past to speed up the typescript-compilation. Unfortunately this setup disables typescript's ability to report the type errors back to webpack, as the errors are silently thrown away in the forked processes stderr. The TS-Plugin for your editor should point them out though. You can also make them visible manually by running typescript via e.g. `npx tsc --noEmit -p tsconfig.json | grep error`.

As we are currently introducing more and more type errors, we should fix that webpack-ts-setup soon as we are effectively not making use of TS's benefits. Also we seem to currently introduce more and more type errors, which will make the effort of fixing them bigger and bigger.

**Comment by GeorgiSTodorov** [comment]
File conflicts.

**Pull request closed without merging**